### PR TITLE
SAA-618: attendance stats list

### PR DESCRIPTION
**Pull request by gerryallan-moj**
Add activity and category details to AllAttendance view

### CODE REVIEW

File: src/main/resources/migrations/common/V2023.06.23__attendance_stats_views.sql [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+       activity.activity_id,
+       activity.summary,
+       activity_category.name
+FROM activity,
+     activity_schedule,
+     prison_regime,
+     scheduled_instance,
+     activity_category
+WHERE activity.activity_id = activity_schedule.activity_id AND activity.prison_code::text = prison_regime.prison_code::text AND activity_schedule.activity_schedule_id = scheduled_instance.activity_schedule_id AND scheduled_instance.start_time = prison_regime.pm_start AND activity.activity_category_id = activity_category.activity_category_id
+UNION
```
Comment (timharrison-moj) [inline src/main/resources/migrations/common/V2023.06.23__attendance_stats_views.sql:28]: What does the `::text` appendage do in this context? Is it needed here? TEXT is a variable length type, so not so sure this is required?

File: src/main/resources/migrations/common/V2023.06.26__attendance_stats_views.sql [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+       ts.prison_code,
+       ts.activity_id,
+       ts.summary,
+       ts.name AS category_name,
+       a.prisoner_number,
+       a.recorded_time
+FROM scheduled_instance si
+         JOIN attendance a ON si.scheduled_instance_id = a.scheduled_instance_id
+         JOIN v_activity_time_slot ts ON si.scheduled_instance_id = ts.scheduled_instance_id
+         LEFT JOIN attendance_reason ar ON a.attendance_reason_id = ar.attendance_reason_id;
```
Comment (timharrison-moj) [inline src/main/resources/migrations/common/V2023.06.26__attendance_stats_views.sql:59]: Is anything using these views yet?  I really like the idea of using views for attendance as it is massively more efficient to narrow down to a single scheduled instance and get all related data in one go.

**Review by timharrison-moj — APPROVED** [review]
LGTM. Couple of questions only.

File: src/main/resources/migrations/common/V2023.06.26__attendance_stats_views.sql (hunk 2 above)
Comment (gerryallan-moj) [reply src/main/resources/migrations/common/V2023.06.26__attendance_stats_views.sql:59]: Yes we are using the views in the attendance stats page

File: src/main/resources/migrations/common/V2023.06.23__attendance_stats_views.sql (hunk 1 above)
Comment (gerryallan-moj) [reply src/main/resources/migrations/common/V2023.06.23__attendance_stats_views.sql:28]: removed

**Review by timharrison-moj — APPROVED** [review]
LGTM.

**Pull request merged**